### PR TITLE
Add `useHover` hook in `VaSidebarItem`

### PR DESCRIPTION
**Pull request by RVitaly1978**
## Description
- [x] add `useHover` hook

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/ui/src/components/va-sidebar/VaSidebarItem/VaSidebarItem.vue [hunk 1]
```diff
@@ -20,17 +19,13 @@
...
+import { useHover } from '../../../composables/useHover'
 
 export default defineComponent({
   name: 'VaSidebarItem',
+
+  inheritAttrs: false,
+
   props: {
     to: {
       type: [String, Object] as PropType<string | Record<string, any>>,
```
Comment (LighthouseKeeperYN) [inline packages/ui/src/components/va-sidebar/VaSidebarItem/VaSidebarItem.vue:31]:
```suggestion
      type: [String, Object] as PropType<RouteLocationRaw>,
```
Import type from 'vue-router'

**Review by LighthouseKeeperYN — APPROVED** [review]

File: packages/ui/src/components/va-sidebar/VaSidebarItem/VaSidebarItem.vue (hunk 1 above)
Comment (RVitaly1978) [reply packages/ui/src/components/va-sidebar/VaSidebarItem/VaSidebarItem.vue]: done

File: packages/ui/src/components/va-sidebar/VaSidebarItem/VaSidebarItem.vue [hunk 2]
```diff
@@ -42,10 +37,12 @@ export default defineComponent({
     hoverColor: { type: String, default: undefined },
     borderColor: { type: String, default: undefined },
   },
+
   setup (props) {
-    const { isHovered, onMouseEnter, onMouseLeave } = useHover()
-    const { getColor, getHoverColor, getTextColor, getFocusColor } = useColors()
+    const anchor = ref<HTMLAnchorElement | undefined>()
```
[the remark this answers is not in the document]
Comment (RVitaly1978) [reply packages/ui/src/components/va-sidebar/VaSidebarItem/VaSidebarItem.vue]: done

File: packages/ui/src/components/va-sidebar/VaSidebarItem/VaSidebarItem.vue [hunk 3]
```diff
@@ -58,11 +56,9 @@ export default defineComponent({
       }
 
       if (props.active) {
-        const border = props.borderColor === undefined ? props.activeColor : props.borderColor
-
-        style['border-color'] = getColor(border)
+        style['border-color'] = getColor(props.borderColor === undefined ? props.activeColor : props.borderColor)
```
Comment (rustem-nasyrov) [inline packages/ui/src/components/va-sidebar/VaSidebarItem/VaSidebarItem.vue:59]: `props.borderColor === undefined` can be replaced with `!props.borderColor` if there is no sense in `undefined`.

Comment (RVitaly1978) [reply packages/ui/src/components/va-sidebar/VaSidebarItem/VaSidebarItem.vue:59]: User can pass in `props.borderColor` an empty string and `getColor` can process an empty string. So, I think, we have to check if `props.borderColor` doesn't define with user.

**Review by rustem-nasyrov — APPROVED** [review]

**Review by m0ksem — APPROVED** [review]

**Pull request merged**